DigitalLearningSolutions.Data/DataServices/FaqsDataService.cs

```diff
@@ -51,5 +53,10 @@ public IEnumerable<Faq> GetPublishedFaqsForTargetGroup(int targetGroup)
                 new { targetGroup }
             );
         }
+
+        public IEnumerable<Faq> GetAllFaqs()

```

Comment:
I originally missed a trick here when I wrote the `GetPublishedFaqsForTargetGroup` method in the DataServices.

I would be tempted to line up with the coding conventions that are now outlined in the swiki: "DataServices to be as generic and reusable as possible. Let the services do the logic".

In this case it would mean:
- Dropping the method `GetPublishedFaqsForTargetGroup`
- Then in the service layer the corresponding method would change to use your new method of `GetAllFaqs` and do the `Publish` and `TargetGroup` filtering.

Some may argue this results in an application taking more than it needs from the DB, which is a valid point. 
But in this instance the `FAQ` dataset is relatively small so the overhead is rather negligable.

Reply:
cool moved the filtering to the service and the tests of the filtering to the service tests